src/Managers/DocumentManager.php

```diff
@@ -78,13 +78,33 @@ public function list(
                 )
             ]);
 
-            for ($i = 0; $i < $result->count(); $i++) {
-                $items[] = $this->getMarshaler()->unmarshalItem($result->get('Items')[$i]);
+            $marshaledItems = $result->get('Items');
+
+            foreach ($marshaledItems as $item) {
+                $items[] = $this->getMarshaler()->unmarshalItem($item);
             }
 
             return $items;
         } catch (DynamoDbException $exception) {
             throw new DocumentQueryFailedException('Failed to query document.', null, $exception);
         }
     }
+
```

Comment:
Somewhat not related to this ticket - but why is this class using inheritance instead of composition?

Reply:
Because of high coupling I had to abstract out `Manager`. 

Made it an injectable class now.